frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1493,15 +1491,10 @@ impl<T: Config> Pallet<T> {
 		let _ = Self::finalize_signed_phase();
 		<QueuedSolution<T>>::take()
 			.map_or_else(
-				|| match T::Fallback::get() {
-					FallbackStrategy::OnChain => Self::onchain_fallback()
-						.map(|s| {
-							// onchain election incurs maximum block weight
-							Self::register_weight(T::BlockWeights::get().max_block);
```

Comment:
so I guess the on chain implementations are expected to register weight themselves now isn't it ?

Reply:
yup, I need to bring back https://github.com/paritytech/substrate/pull/9569/files#diff-9e46355d63e0cdccfac92cb606bfdb8425664d91d225a7cf990c9bbab5f7692cL58